GeneralizedMorphismsForCAP/gap/GeneralizedMorphismCategoryByCospans.gi

```diff
@@ -376,6 +376,130 @@ InstallMethod( AsGeneralizedMorphismByCospan,
     
 end );
 
+####################################
+##
+## Constructors of lifts of exact functors
+##
+####################################
+
+##
+InstallMethod( AsGeneralizedMorphismByCospan,
+        [ IsCapFunctor, IsString, IsCapCategory, IsCapCategory ],
+
+  function( F, name, A, B )
+    local gmcF;
+    
+    if not IsIdenticalObj( AsCapCategory( Source( F ) ), UnderlyingHonestCategory( A ) ) then
+        Error( "the source of the functor and the category underlying the source category with ambient object do not coincide\n" );
+    elif not IsIdenticalObj( AsCapCategory( Range( F ) ), UnderlyingHonestCategory( B ) ) then
+        Error( "the target of the functor and the category underlying the target category with ambient object do not coincide\n" );
+    fi;
+    
+    gmcF := CapFunctor( name, A, B );
+    
+    AddObjectFunction( gmcF,
+            function( obj )
+              return GeneralizedMorphismByCospansObject( ApplyFunctor( F, UnderlyingHonestObject( obj ) ) );
+            end );
+    
+    AddMorphismFunction( gmcF,
+            function( new_source, mor, new_range )
+              return GeneralizedMorphismByCospan( ApplyFunctor( F, Arrow( mor ) ), ApplyFunctor( F, ReversedArrow( mor ) ) );
+            end );
+    
+    gmcF!.UnderlyingFunctor := F;
```

Comment:
Maybe it is a good idea to realize `AsGeneralizedMorphismByCospan` 
as an attribute of a Cap functor.
Then we could do the same for its counterpart on natural transformations
(see below)
and get rid of this `UnderlyingFunctor`.